packages/salesforcedx-sobjects-faux-generator/src/generator/typingGenerator.ts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import * as fs from 'fs';
+import { EOL } from 'os';
+import * as path from 'path';
+import { rm } from 'shelljs';
+import { SObject } from '../types';
+import { DeclarationGenerator, FieldDeclaration } from './declarationGenerator';
+
+export class TypingGenerator {
+  private declGenerator: DeclarationGenerator;
+
+  public constructor() {
+    this.declGenerator = new DeclarationGenerator();
+  }
+
+  public generate(sobjects: SObject[], targetFolder: string): void {
+    if (!fs.existsSync(targetFolder)) {
+      fs.mkdirSync(targetFolder);
+    }
+
+    for (const sobject of sobjects) {
+      if (sobject.name) {
+        this.generateTypingForObject(targetFolder, sobject);
+      }
+    }
+  }
+
+  public generateTypingForObject(folderPath: string, sobject: SObject): string {
+    const typingPath = path.join(folderPath, sobject.name + '.d.ts');
+    if (fs.existsSync(typingPath)) {
```

Comment:
this isn't needed as `fs.writeFileSync` overwrites the file if it exists

Reply:
I was getting a permission error before I added this, probably since the perms were set to readonly when the previous contents were written?

Reply:
Interesting, my definition files have permission `644`. Yeah it makes sense to remove first if we are writing with permission `444`.